rpc/handlers_test.go

```diff
@@ -1742,194 +1742,54 @@ func TestEvents(t *testing.T) {
+		assert.Equal(t, jsonrpc.InvalidJSON, err.Code)
+	})
 
-		expectedInvokeResp := &rpc.AddInvokeTxResponse{
-			TransactionHash: new(felt.Felt).SetBytes([]byte("random")),
+	t.Run("ok response", func(t *testing.T) {
+		mockGateway.EXPECT().AddTransaction(gomock.Any()).Return(json.RawMessage(`
+		{
+			"transaction_hash" : "0x1",
+			"contract_address" : "0x2",
+			"class_hash" : "0x3"
 		}
-		expectedInvokeRespB, err := json.Marshal(expectedInvokeResp)
-		require.NoError(t, err)
-
-		mockGateway.EXPECT().AddTransaction(invokeTxByte).Return(expectedInvokeRespB, nil)
+		`), nil)
 
-		resp, handlerErr := handler.AddInvokeTransaction(invokeTxByte)
-		require.Nil(t, handlerErr)
-		assert.Equal(t, expectedInvokeResp.TransactionHash, resp.TransactionHash)
+		response, err := handler.AddTransaction(json.RawMessage(`{}`))
+		require.Nil(t, err)
+		assert.Equal(t, "0x1", response.TransactionHash.String())
+		assert.Equal(t, "0x2", response.ContractAddress.String())
+		assert.Equal(t, "0x3", response.ClassHash.String())
 	})
-}
-
-func TestAddDeclareTransaction(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	t.Cleanup(mockCtrl.Finish)
-
-	mockGateway := mocks.NewMockGateway(mockCtrl)
-	log := utils.NewNopZapLogger()
-	handler := rpc.New(nil, nil, utils.MAINNET, mockGateway, nil, nil, "", log)
 
-	t.Run("correct V2", func(t *testing.T) {
+	t.Run("compresses sierra program", func(t *testing.T) {
 		declareTxV2 := `{"contract_class":{"sierra_program":["0x0","0x0"]},"type":"DECLARE","version":"0x2"}`
 		gwDeclareTxV2 := `{"contract_class":{"sierra_program":"H4sIAAAAAAAA/4pWMqgwUNIBk7GAAAAA//9n6XuWDQAAAA=="},"type":"DECLARE","version":"0x2"}`
 
-		tx := json.RawMessage(declareTxV2)
-		expectedDeclareResp := &rpc.DeclareTxResponse{
-			TransactionHash: new(felt.Felt).SetBytes([]byte("randomrandom")),
-			ClassHash:       new(felt.Felt).SetBytes([]byte("randomrandom")),
-		}
-		expectedDeclareRespB, err := json.Marshal(expectedDeclareResp)
-		require.NoError(t, err)
-
-		mockGateway.EXPECT().AddTransaction(json.RawMessage(gwDeclareTxV2)).Return(expectedDeclareRespB, nil)
-
-		resp, handlerErr := handler.AddDeclareTransaction(tx)
-
-		require.Nil(t, handlerErr)
-		assert.Equal(t, *expectedDeclareResp, *resp)
-	})
-
-	t.Run("correct V1", func(t *testing.T) {
-		declareTxV1 := `{"type":"DECLARE","version":"0x1","contract_class":{"sierra_program":"H4sIAAAAAAAA/1IyqDBQAgQAAP//Jz3M3QUAAAA="}}`
-		tx := json.RawMessage(declareTxV1)
-		expectedDeclareResp := &rpc.DeclareTxResponse{
-			TransactionHash: new(felt.Felt).SetBytes([]byte("randomrandom")),
-			ClassHash:       new(felt.Felt).SetBytes([]byte("randomrandom")),
-		}
-		expectedDeclareRespB, err := json.Marshal(expectedDeclareResp)
-		require.NoError(t, err)
-
-		mockGateway.EXPECT().AddTransaction(json.RawMessage(declareTxV1)).Return(expectedDeclareRespB, nil)
-
-		resp, handlerErr := handler.AddDeclareTransaction(tx)
-
-		require.Nil(t, handlerErr)
-		assert.Equal(t, *expectedDeclareResp, *resp)
-	})
```

Comment:
Can you please explain why you are removing these test cases?

Reply:
They provide no additional coverage, both in terms of code coverage or edge case coverage.

AddTransaction works on raw JSON. So it only makes sense to test it for the modifications that it does on the json (ie change invoke type and compress sierra classes) and how it responds to invalid json.

Testing it for all the separate transactions that are just relayed without any modifications is just not necessary because the API of AddTransaction makes no distinction on them.